src/main/ApplicationImpl.cpp

```diff
@@ -609,6 +609,39 @@ ApplicationImpl::syncAllMetrics()
     syncOwnMetrics();
 }
 
+template <typename T>
+std::shared_ptr<T>
+castMetric(std::shared_ptr<medida::MetricInterface> metric)
+{
+    return std::dynamic_pointer_cast<T>(metric);
+}
+
+void
+ApplicationImpl::clearMetrics(const std::string& domain)
+{
+    auto const& metrics = mMetrics->GetAllMetrics();
+    for (auto const& kv : metrics)
+    {
+        auto metricName = kv.first;
+        if (domain.empty() || metricName.domain() == domain)
+        {
+            auto metric = kv.second;
+            if (auto counter = castMetric<medida::Counter>(metric))
```

Comment:
This should be replaced with `medida::MetricProcessor` and `medida::MetricInterface::Process` in order to avoid the use of dynamic cast.